app/sqaws.py

```diff
@@ -127,7 +136,20 @@ def build_instance_model(pricing: PricingData, region_name: str, instance_dict:
                     stop_after=stop_after,
                     terminate_after=terminate_after,
                     contact=contact,
-                    nagbot_state=nagbot_state)
+                    nagbot_state=nagbot_state,
+                    stop_after_str=stop_after_str,
+                    terminate_after_str=terminate_after_str)
+
+
+# Find 'stop after' and 'terminate after' fields in an EC2 instance, regardless of formatting
+def get_stop_and_terminate_after(tags: dict) -> tuple[str, str]:
+    stop_after, terminate_after = '', ''
```

Comment:
Sometimes new instances get created without any tags; but nagbot still needs to be able to create the tag. I think here we could specify the default tags that would just work. I'd vote for "StopAfter" and "TerminateAfter" as the default.